### PR TITLE
x/txfee: new extension

### LINKED ISSUE #1091: Implement Transaction size fee
There should be an additional transaction fee that charges for the amount of bytes of the signed transaction.

In https://github.com/iov-one/transaction-size-fee/blob/master/simulate.py#L3-L10 there is the algorithm we discussed.

`atomic` is an appreviation for "atomic units [of IOV tokens]", i.e. 1 atomic is 0.000000001 IOV. All calculations should be done on integers of atomic units. With atomics stored as uint64 we can do safe math up to ~18 billion IOV, which should be sufficient for the transaction sizes we can expect.

With small modifications in the pythong script from above we can generate a nice set of test vectors.

The two constants `FREE_CONTINGENT` and `MULTIPLIER` should be configurable such that they can be changed over time.

### PR DESCRIPTION
**Pull request by husio**
A new extension `x/txfee` allows to configure an additional transaction
fee. Additional fee amount depends on the transaction binary size.
`x/txfee` is enabled in `cmd/bnsd` and can be confiured via `gcond`
based confiuration.

`cmd/bnscli` was updated to support new extension.

Batch transaction fee is computed only once for the original transaction
and not for each message that batch contains.

resolve #1091

### CODE REVIEW

File: cmd/bnscli/clitests/txfee_print_rates.test.gold [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+60026 bytes	4448148879274931250 IOV
+62501 bytes	4822530820312500000 IOV
+65026 bytes	5220062452497431250 IOV
+67601 bytes	5641678961006400000 IOV
+70226 bytes	6088334049534431250 IOV
+72901 bytes	6560999940294900000 IOV
+75626 bytes	7060667374019531250 IOV
+78401 bytes	7588345609958400000 IOV
+81226 bytes	8145062425879931250 IOV
+84101 bytes	8731864118070900000 IOV
```
Comment (orkunkl) [inline cmd/bnscli/clitests/txfee_print_rates.test.gold]: `8731864118070900000` woah this looks a costly tx. Are you sure about the rates?

Comment (husio) [reply cmd/bnscli/clitests/txfee_print_rates.test.gold]: This is a test that shows how the prices are growing. With base fee `1234567890 IOV`, this is an expected cost :grin:

File: x/txfee/decorator.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	return TransactionFee(txSize, conf.BaseFee, conf.FreeBytes)
+}
+
+func TransactionFee(txSize int, baseFee coin.Coin, freeBytes int32) (*coin.Coin, error) {
+	paidSize := int32(txSize) - freeBytes
+	if paidSize <= 0 {
+		return coin.NewCoinp(0, 0, "IOV"), nil
+	}
+
+	// ((max(0, bytes_size(tx) - free_bytes) ** 2) * base_fee
```
Comment (husio) [inline x/txfee/decorator.go]: @davepuchyr can you double check that this is what is expected? You can [use `bnscli txfee-print-rates` command](https://github.com/iov-one/weave/pull/1094/files/fb72b7195db979a5f69c208bd5c1a83b539067cc#diff-2820dd108426b29797709620e8153ce7) to display price rates for any given configuration.

Comment (davepuchyr) [reply x/txfee/decorator.go]: LGTM

**Pull request merged**